src/scripts/clipperUI/components/pdfClipOptions.tsx

```diff
@@ -96,30 +106,62 @@ class PdfClipOptionsClass extends ComponentBase<PdfClipOptionsState, ClipperStat
 	getPageRangeRadioElement(): any {
 		let pdfPreviewInfo = this.props.clipperState.pdfPreviewInfo;
 
-		let pageRangeIsInvalid = this.invalidPageRange();
-		let invalidClassName = pageRangeIsInvalid ? "invalid" : "";
+		let invalidClassName = pdfPreviewInfo.shouldShowPopover ? "invalid" : "";
 		return (
 			<div id={Constants.Ids.radioPageRangeLabel} class="pdf-control" {...this.enableInvoke(this.onSelectionChange, 61, false) }>
 				<div class="pdf-indicator pdf-radio-indicator">
 					{!pdfPreviewInfo.allPages ? <div class={Constants.Classes.radioIndicatorFill}></div> : ""}
 				</div>
 				{!pdfPreviewInfo.allPages ?
-					<input type="text" id={Constants.Ids.rangeInput} class={invalidClassName} placeholder="e.g. 1-5, 7, 9-12" value={this.props.clipperState.pdfPreviewInfo.selectedPageRange} {...this.enableInvoke(this.onSelectionChange, 62, false) }></input>
+					<input
+						type="text"
+						id={Constants.Ids.rangeInput}
+						class={invalidClassName}
+						config={this.handlePopoverLifeCycle.bind(this)}
+						placeholder="e.g. 1-5, 7, 9-12"
+						onfocus={this.onTextInputFocus.bind(this)}
+						value={this.props.clipperState.pdfPreviewInfo.selectedPageRange} {...this.enableInvoke(this.onSelectionChange, 62, false) }>
+					</input>
 					: <span class="pdf-label">{Localization.getLocalizedString("WebClipper.Preview.Header.PdfPageRangeRadioButtonLabel")}</span>}
-				{!pdfPreviewInfo.allPages && pageRangeIsInvalid ?
-					<div class="popover">{Localization.getLocalizedString("WebClipper.Preview.Header.PdfInvalidPageRange")}</div>
-					: ""}
 			</div>
 		);
 	}
 
-	private invalidPageRange(): boolean {
-		let pagesToShow = StringUtils.parsePageRange(this.props.clipperState.pdfPreviewInfo.selectedPageRange);
-		let validUpperBounds = this.props.clipperState.pdfResult.status === Status.Succeeded ? _.every(pagesToShow, (ind: number) => {
-			return ind <= this.props.clipperState.pdfResult.data.get().pdf.numPages();
-		}) : true;
+	// Destroy the currently existing popover
+	// Then create a new one if necessary
+	// TODO: make this into an actual component
+	private handlePopoverLifeCycle() {
```

Comment:
Right now we're using a static field to maintain the popover, and we expose a lot of low-level methods such as destroy() which can be scary. Additionally, I don't think PdfClipOptions should have to worry about popover lifecycles (admittedly, we can definitely improve our MVC-logic throughout our codebase :P ). I recommend a couple of things:

a) Wrap the popover object in a class that implements an interface that exposes some high-level methods. Example methods could be show(), hide(), setText() etc.
b) Have a popover manager class where you can pass in a string key and get back the popover object associated with it. Creation of popovers can also be done through this class. That way, we can ensure that we don't have duplicate popovers, and at the same time, we can scale popovers to multiple use-cases in the future, all while keeping things testable.